src/Workspaces/Core/Portable/FindSymbols/SyntaxTree/SyntaxTreeIndex_Create.cs

```diff
@@ -70,6 +70,7 @@ internal sealed partial class SyntaxTreeIndex
                 var containsGlobalSuppressMessageAttribute = false;
                 var containsConversion = false;
                 var containsGlobalKeyword = false;
+                var containsIdentifierContainerPreprocessorDirective = false;
```

Comment:
Just check if the root has the ContainsDirective bit set.

Reply:
Isn't this bit of code for determining the bit flag of whether pp directives are contained?

Reply:
The root already had a bit telling you this. Just use that. Root.ContainsDirective

Reply:
The reason why I opted in for one more flag is that `ContainsDirective` also includes `#region` and `#endregion` directives, which is a very commonly-used practice, and in most cases people don't use pp symbols. The filter still filters out documents not containing the pp symbol identifier which will rarely conflict with a non-pp one causing a false positive, so that's not too bad I suppose.

Reply:
yes.  once we filter to both name and 'has directives', we very likely have the set of documents of interest.